File: Python/flashmen/basesolve/g.py
def solve():
    a, b, c, d = 0, *map(int, input().split())
    bit, f = 1, 1

    for i in range(62):
        bit_b = 1 if (b & bit) else 0
        bit_c = 1 if (c & bit) else 0
        bit_d = 1 if (d & bit) else 0

        if (bit_b and not bit_c and not bit_d) or (not bit_b and bit_c and bit_d):
            f = 0
            break

        a |= (bit * (1 - bit_d)) if (bit_b and bit_c) else bit * bit_d
        bit <<= 1

    print(a if f else -1)


if __name__ == "__main__":
    t = int(input().strip())
    for _ in range(t):
        solve()

File: cpp/flashmen/basic/g.cpp
// Copyright 2024 Allergy
/* @author    Allergy
 * @email     Allergy527@gmail.com
 * @workspace cpp\flashmen\g.cpp
 * @date      2024/10/26 18:07:10
 */
#include <bits/stdc++.h>
#define lowbit(x) ((x) & (-(x)))
#define ln '\n'
#define int int64_t

using namespace std;  // NOLINT
using ll = int64_t;
int inf = 0x3f3f3f3f;
ll ll_inf = 0x3f3f3f3f3f3f3f3f;

void solve() {
    int a = 0, b, c, d, bit = 1, f = 1;
    cin >> b >> c >> d;
    for (int i = 0; i < 62; ++i) {
        int bit_b = (b & bit) ? 1 : 0;
        int bit_c = (c & bit) ? 1 : 0;
        int bit_d = (d & bit) ? 1 : 0;
        if (bit_b && (!bit_c) && (!bit_d) || (!bit_b) && bit_c && bit_d) {
            f = 0;
            break;
        }
        a |= (bit_b && bit_c) ? bit * (1ll - bit_d) : bit * bit_d;
        bit <<= 1;
    }
    cout << (f ? a : -1) << ln;
}

signed main() {
    ios::sync_with_stdio(false), cin.tie(0), cout.tie(0);
    int t = 1;
    cin >> t;
    while (t--) solve();
    cin >> t;
    return 0;
}
